src/apps/backend/modules/booking/rest-api/booking-controller.ts

```diff
@@ -0,0 +1,30 @@
+import { applicationController, Request, Response } from '../../application';
+import { HttpStatusCodes } from '../../http';
+import BookingService from '../booking-service';
+import { BookingCustomRequestBody } from '../types';
+
+export class BookingController {
+  createBooking = applicationController(
+    async (req: Request<BookingCustomRequestBody>, res: Response) => {
+      const toolCall = req.body.message.toolCalls[0];
+      const { firstName, phoneNumber, address, availability } =
+        toolCall.function.arguments;
+
+      const booking = await BookingService.createBooking({
+        firstName,
+        phoneNumber,
+        address,
+        availability,
+      });
+
+      res.status(HttpStatusCodes.CREATED).send({
```

Comment:
We need to return a `Booking` instance, with atleast a booking reference # so Voice AI agent can let the customer know of it.

Reply:
https://docs.vapi.ai/tools-calling#server-response-format-providing-results-and-context
This needs to be in this following format. According to this, we can send booking instance as a result but we can't send just booking instance.

Since we are just using mock data and we are not actually creating booking, should we pass random reference id or we can do this after integration with Housecall Pro.

Reply:
See this [comment](https://jalantechnology.sharepoint.com/:w:/s/HVACAgent/ERv6tLDmss1EpOit0OYvwqMBhGuARgjlSDXCblx5oJZrTg?e=tC6gUv&nav=eyJjIjoxMzc3MjUwMDA0fQ) on how to best handle this